src/python/python-proto/python_proto/tests/strategies.py

```diff
@@ -57,23 +93,71 @@ def uuids(
     return st.builds(Uuid, lsb=lsbs, msb=msbs)
 
 
+def timestamps(
+    seconds: st.SearchStrategy[int] = st.integers(
+        min_value=TIMESTAMP_SECONDS_MIN, max_value=TIMESTAMP_SECONDS_MAX
+    ),
+    nanos: st.SearchStrategy[int] = st.integers(
+        min_value=TIMESTAMP_NANOS_MIN,
+        max_value=TIMESTAMP_NANOS_MAX,
+    ),
+) -> st.SearchStrategy[Timestamp]:
+    return st.builds(Timestamp, seconds=seconds, nanos=nanos)
+
+
+def durations(
+    seconds: st.SearchStrategy[int] = st.integers(
+        min_value=DURATION_SECONDS_MIN, max_value=DURATION_SECONDS_MAX
+    ),
+    nanos: st.SearchStrategy[int] = st.integers(
+        min_value=DURATION_NANOS_MIN,
+        max_value=DURATION_NANOS_MAX,
+    ),
+) -> st.SearchStrategy[Duration]:
+    return st.builds(Duration, seconds=seconds, nanos=nanos)
+
+
+#
+# pipeline
+#
+
+
 def metadatas(
     trace_ids: st.SearchStrategy[uuid.UUID] = st.uuids(),
     tenant_ids: st.SearchStrategy[uuid.UUID] = st.uuids(),
+    event_source_ids: st.SearchStrategy[uuid.UUID] = st.uuids(),
+    created_times: st.SearchStrategy[datetime.datetime] = st.datetimes(),
+    last_updated_times: st.SearchStrategy[datetime.datetime] = st.datetimes(),
 ) -> st.SearchStrategy[Metadata]:
-    return st.builds(Metadata, trace_id=trace_ids, tenant_id=tenant_ids)
+    return st.builds(
+        Metadata,
+        trace_id=trace_ids,
+        tenant_id=tenant_ids,
+        event_source_id=event_source_ids,
+        created_time=created_times,
+        last_updated_time=last_updated_times,
+    )
+
+
+def raw_logs(
+    log_events: st.SearchStrategy[bytes] = st.binary(
+        min_size=MIN_LOG_EVENT_SIZE, max_size=MAX_LOG_EVENT_SIZE
+    )
+) -> st.SearchStrategy[RawLog]:
+    return st.builds(RawLog, log_event=log_events)
 
 
 def envelopes(
     metadatas: st.SearchStrategy[Metadata] = metadatas(),
-    inner_messages: st.SearchStrategy[bytes] = st.binary(),
-    inner_types: st.SearchStrategy[str] = st.text(),
+    inner_messages: st.SearchStrategy[SerDe] = uuids()
+    | timestamps()
+    | durations()
+    | raw_logs(),
```

Comment:
TIL SearchStrategy supports `__or__`

Reply:
yeah it's very convenient and expressive